File: tests/python/test_providers.py
import unittest
from mini_langchain import (
    Chain, AgentExecutor, PromptTemplate,
    SambaNovaLLM, OpenAILLM, AnthropicLLM, GoogleGenAILLM, OllamaLLM
)

class TestProviders(unittest.TestCase):
    def setUp(self):
        self.prompt = PromptTemplate("Hello {input}", ["input"])

    def test_sambanova_initialization(self):
        # API Key optional in some contexts or mock
        llm = SambaNovaLLM("Meta-Llama-3-8B", "test-key")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)

    def test_openai_initialization(self):
        llm = OpenAILLM("sk-test", "gpt-4o")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)
        # Check Agent
        agent = AgentExecutor(llm)
        self.assertIsNotNone(agent)

    def test_anthropic_initialization(self):
        llm = AnthropicLLM("sk-ant-test", "claude-3-opus")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)

    def test_google_initialization(self):
        llm = GoogleGenAILLM("AIza-test", "gemini-pro")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)

    def test_ollama_initialization(self):
        llm = OllamaLLM("llama3")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)

    def test_open_router_via_openai(self):
        # OpenRouter uses OpenAI client with custom URL
        llm = OpenAILLM("sk-or-test", "nous/hermes", "https://openrouter.ai/api/v1")
        chain = Chain(self.prompt, llm)
        self.assertIsNotNone(chain)
    
if __name__ == '__main__':
    print("Running Provider Tests...")
    unittest.main()

File: tests/node/test_providers.js
const {
    Chain, AgentExecutor, PromptTemplate,
    SambaNovaLLM, OpenAILLM, AnthropicLLM, GoogleGenAILLM, OllamaLLM
} = require('../../node/index.js');
const assert = require('assert');

console.log("--- Testing Node.js Providers Initialization ---");

const prompt = new PromptTemplate("Hello {input}", ["input"]);

// 1. SambaNova
try {
    const llm = new SambaNovaLLM("Meta-Llama-3", "key");
    assert(llm, "SambaNovaLLM should be created");
    const chain = new Chain(prompt, llm);
    assert(chain, "Chain should accept SambaNovaLLM");
    console.log("PASS: SambaNovaLLM");
} catch (e) {
    console.error("FAIL: SambaNovaLLM", e);
    process.exit(1);
}

// 2. OpenAI
try {
    const llm = new OpenAILLM("sk-key", "gpt-4");
    assert(llm, "OpenAILLM should be created");
    const chain = new Chain(prompt, llm);
    assert(chain, "Chain should accept OpenAILLM");
    // Test Agent acceptance
    const agent = new AgentExecutor(llm);
    assert(agent, "Agent should accept OpenAILLM");
    console.log("PASS: OpenAILLM");
} catch (e) {
    console.error("FAIL: OpenAILLM", e);
    process.exit(1);
}

// 3. Anthropic
try {
    const llm = new AnthropicLLM("sk-ant-key", "claude-3");
    assert(llm, "AnthropicLLM should be created");
    const chain = new Chain(prompt, llm);
    assert(chain, "Chain should accept AnthropicLLM");
    console.log("PASS: AnthropicLLM");
} catch (e) {
    console.error("FAIL: AnthropicLLM", e);
    process.exit(1);
}

// 4. Google
try {
    const llm = new GoogleGenAILLM("key", "gemini-pro");
    assert(llm, "GoogleGenAILLM should be created");
    const chain = new Chain(prompt, llm);
    assert(chain, "Chain should accept GoogleGenAILLM");
    console.log("PASS: GoogleGenAILLM");
} catch (e) {
    console.error("FAIL: GoogleGenAILLM", e);
    process.exit(1);
}

// 5. Ollama
try {
    const llm = new OllamaLLM("llama3");
    assert(llm, "OllamaLLM should be created");
    const chain = new Chain(prompt, llm);
    assert(chain, "Chain should accept OllamaLLM");
    console.log("PASS: OllamaLLM");
} catch (e) {
    console.error("FAIL: OllamaLLM", e);
    process.exit(1);
}

console.log("--- All Node.js Providers Passed ---");
